### DISCUSSION TITLE
Composite HID and Vendor class

### DISCUSSION THREAD

**Discussion by alexpa940** [hathach/tinyusb#2831] [Q&A] [answered]
Hi,
I'm going to try to make a composite usb device hid keyboard plus vendor device. 
The device is initialized incorrectly in debian 10.
The microcontroller gets stuck with TU_ASSERT in the usbd.c file.
```
        p_desc += drv_len;

        break; // exit driver find loop
      }
    }

    // Failed if there is no supported drivers
    TU_ASSERT(drv_id < TOTAL_DRIVER_COUNT);
  }

  return true;
}
```

Separately, vendor and hid work correctly.

What am I doing wrong?
I've posted all the code at https://github.com/alexpa940/Miscellaneous/blob/main/usb_descriptors.c
[![call stack](https://github.com/user-attachments/assets/c6717fde-9dbc-40cd-a567-39f3f13ef0ac)](https://github.com/alexpa940/Miscellaneous/blob/main/call%20stack.png)

**Comment by HiFiPhile** [comment]
Have you enabled both class in `tusb_config.h` ?

**Reply by alexpa940** [reply]
Yes.
One CFG_TUD_HID and one CFG_TUD_VENDOR.

**Comment by HiFiPhile** [comment, marked as answer]
Are you using 0.17 release ? It could be https://github.com/hathach/tinyusb/commit/53989a99f60e755094f146e936b49b0ef24e6ec0

**Reply by alexpa940** [reply]
I was on the 0.17.0 tag. It does not include this change. After upgrading to the master branch, the error disappeared.
